### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by pradykaushik**
Using logrus for logging.
Two types of logs.
1. Task ranker specific logs.
2. Task ranking results logs.

Console logging is disabled for the task ranker to prevent logs from getting mixed with the logs of the application that is using it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,3 +148,13 @@ HOST = localhost
 
 #### Tear-Down
 Once finished testing, tear down the test environment by running [`./tear_down_test_env`](./tear_down_test_env).
+
+### Logs
+Task Ranker uses [logrus](https://github.com/sirupsen/logrus) for logging. To prevent logs from Task Ranker
+mixing in with logs from the application that is using it, console logging is disabled.
+There are [two types of logs](./logger/logger.go) as mentioned below. 
+1. Task Ranker logs - These logs are Task Ranker specific and correspond functioning of the library.
```
Comment (SrinidhiRaghavendra) [inline README.md:156]: correspond to*

File: logger/hook.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+package logger
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io"
+)
+
+// WriterHook is a hook that writes logs, that contain the at least one of the specified set of topics
```
Comment (SrinidhiRaghavendra) [inline logger/hook.go:23]: contain at least one*

File: logger/logger.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	// Disabling log to stdout.
+	log.SetOutput(ioutil.Discard)
+	// Setting highest log level.
+	log.SetLevel(logrus.InfoLevel)
+
+	// Creating the log files.
+	now := time.Now()
+	var err error
+
+	taskRankerLogFile, err = createTaskRankerLogFile(now)
```
Comment (SrinidhiRaghavendra) [inline logger/logger.go:78]: taskRankerLogFile pointer already set within the function body. Do we need the assignment again?

File: logger/logger.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	if err != nil {
+		return err
+	}
+	taskRankingResultsLogFile, err = createTaskRankingResultsLogFile(now)
```
Comment (SrinidhiRaghavendra) [inline logger/logger.go:82]: Same as line 78

File: logger/logger.go (hunk 3 above)
Comment (pradykaushik) [reply logger/logger.go:78]: you are right. Currently, however there is one package scoped variable and another local to the function. I'm going to get rid of the local version and just initialize the package scoped variable.

**Pull request merged**